File: server/lib/avstreams.js
const { Readable } = require("stream")
const { AvVideoFrame, AvVideoCapture } = require('../build/Release/libavstreams')
console.debug('Avstreams exports', nativeAvStreams)

class AvVideosCaptureStream extends Readable {
    constructor(options) {
        super(options);
        this.capture = new AvVideoCapture();
    }

    _read() {
        this.push(this.capture.readFrame());
    }
}

module.exports = {
    "AvVideoFrame": AvVideosFrame,
    "AvVideosCaptureStream", AvVideosCaptureStream
}

File: server/ccsrc/avstreams.cc
#include <napi.h>

#include "avvideoframe.h"
#include "avvideocapturestream.h"

Napi::Object InitModule(Napi::Env env, Napi::Object exports) {
    avstreams::AvVideoFrame::DeclareClass(env, exports);
    avstreams::AvVideoCaptureStream::DeclareClass(env, exports);
    return exports;
}

NODE_API_MODULE(libavstreams, InitModule)